sway-core/src/error.rs

```diff
@@ -69,6 +70,12 @@ pub(crate) fn ok<T>(
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct ParseResult<T> {
```

Comment:
The name `ParseResult` sounds to me like a result type for parsing -- i.e. `<T, ParseError>`. Would it be possible to either document this struct and its fields or rename it to something a bit more descriptive?